File: app/service/package_json.py
from typing import Optional


class PackageJSON:
    _raw_data = None

    def __init__(self, data):
        self._raw_data = data

    def get_script(self, name: str) -> Optional[str]:
        """
        Returns the commands specified for the given script.
        This is extracted from the "scripts" object of the package json.
        If the commands for the script are not present, None is returned.
        eg-
          script = "npm run build"
          name = "build"
          package.json = {"scripts": {"build": "babel ."}}
          returns = "babel ."
        """
        pass

    def raw(self) -> str:
        return self._raw_data

File: internal/packagejson/packagejson.go
package packagejson

import (
	"encoding/json"
	"errors"
)

type PackageJSON struct {
	rawData    map[string]interface{}
	rawDataStr string
}

func NewPackageJSON(content string) (*PackageJSON, error) {
	var data map[string]interface{}
	err := json.Unmarshal([]byte(content), &data)
	if err != nil {
		return nil, err
	}
	return &PackageJSON{rawData: data, rawDataStr: content}, nil
}

func (p *PackageJSON) GetScript(name string) (string, error) {
	scripts, ok := p.rawData["scripts"].(map[string]interface{})
	if !ok {
		return "", errors.New("scripts not found in package.json")
	}
	script, ok := scripts[name].(string)
	if !ok {
		return "", nil
	}
	return script, nil
}

func (p *PackageJSON) Raw() map[string]interface{} {
	return p.rawData
}

func (p *PackageJSON) String() string {
	return p.rawDataStr
}
